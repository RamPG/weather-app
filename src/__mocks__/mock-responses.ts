import {
  GetGeoCityNameResponseType,
  GetGeoCoordsResponseType,
  GetWeatherResponseType,
} from '../types/response-types';

export const weatherResponseMock: GetWeatherResponseType = {
  lat: 25.5,
  lon: 24.4,
  timezone: '1200',
  timezone_offset: 3600,
  daily: [
    {
      dt: 5,
      sunrise: 5,
      sunset: 5,
      temp: {
        day: 25,
        min: 20,
        max: 27,
        night: 15,
        eve: 15,
        morn: 15,
      },
      feels_like: {
        day: 25,
        night: 21,
        eve: 23,
        morn: 24,
      },
      pressure: 10,
      humidity: 50,
      dew_point: 1,
      uvi: 1,
      clouds: 1,
      wind_speed: 34,
      wind_deg: 21,
      weather: [
        {
          id: 1,
          main: 'Clouds',
          icon: '04n',
        },
      ],
      pop: 25,
      rain: 15,
    },
  ],
  current: {
    dt: 5,
    sunrise: 5,
    sunset: 5,
    temp: 25,
    feels_like: 23,
    pressure: 10,
    humidity: 50,
    dew_point: 1,
    uvi: 1,
    clouds: 1,
    visibility: 1,
    wind_speed: 34,
    wind_deg: 21,
    weather: [
      {
        id: 1,
        main: 'Clouds',
        icon: '04n',
      },
    ],
  },
};

export const coordsResponseMock: GetGeoCoordsResponseType = {
  place_id: '1',
  licence: '1',
  osm_type: '1',
  osm_id: '1',
  boundingbox: ['1'],
  lat: '25.5',
  lon: '24.4',
  display_name: 'Moscow, Russia',
  class: '1',
  type: '1',
  importance: 1,
  icon: '1',
};

export const cityNameResponseMock: GetGeoCityNameResponseType = {
  place_id: '1',
  licence: '1',
  osm_type: '1',
  osm_id: '1',
  boundingbox: ['1'],
  lat: '25.5',
  lon: '24.4',
  display_name: 'Moscow, Russia',
  address: {
    city: 'Moscow',
    county: 'Moscow District',
    state: '',
    country: 'Russia',
    counry_code: '05',
  },
};
