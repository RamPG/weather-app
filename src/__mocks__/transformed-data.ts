import { DataCoordsStateType, TransformedCurrentDataType, TransformedDailyDataType } from '../types/state-types';

export const transformedDataCoords: DataCoordsStateType = {
  location: 'Moscow',
  latitude: 25.5,
  longitude: 24.4,
};

export const transformedDataCurrent: TransformedCurrentDataType = {
  imgLink: 'https://openweathermap.org/img/wn/04n@2x.png',
  temp: '-249째C',
  feelsLike: '-251째C',
  humidity: 50,
  weather: 'Clouds',
  windSpeed: 34,
};

export const transformedDataDaily: Array<TransformedDailyDataType> = [
  {
    id: 0,
    imgLink: 'https://openweathermap.org/img/wn/04n@2x.png',
    weekDayName: 'Friday',
    monthDay: 14,
    monthDayName: 'August',
    temp: {
      day: '-249째C',
      night: '-259째C',
    },
    weather: 'Clouds',
  },
];
