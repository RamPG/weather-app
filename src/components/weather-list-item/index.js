import { WeatherListItem } from './weather-list-item';

export { WeatherListItem };
