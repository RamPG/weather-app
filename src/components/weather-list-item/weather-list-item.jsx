import React from 'react';

import './weather-list-item.scss';

const WeatherListItem = ({
  weekDayName, monthDay, monthDayName, temp, weather, imgLink,
}) => (
  <>
    <p className="weather-list__item-week-day">
      {weekDayName}
    </p>
    <p className="weather-list__item-date">
      {monthDay}
      {' '}
      {monthDayName}
    </p>
    <p className="weather-list__item-img-block">
      <img
        className="weather-list__weather-img"
        src={imgLink}
        width="100"
        height="100"
        alt="weather"
      />
    </p>
    <p className="weather-list__item-temp-day">
      {temp.day}
    </p>
    <p className="weather-list__item-temp-night">
      {temp.night}
    </p>
    <p className="weather-list__item-weather">
      {weather}
    </p>
  </>
);

export default WeatherListItem;
