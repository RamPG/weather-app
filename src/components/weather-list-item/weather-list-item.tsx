import React, { FunctionComponent } from 'react';

import './weather-list-item.scss';

import { TransformedDailyDataType } from '../../types/state-types';

export const WeatherListItem: FunctionComponent<TransformedDailyDataType> = ({
  weekDayName, monthDay, monthDayName, temp, weather, imgLink,
}) => (
  <>
    <p>
      {weekDayName}
    </p>
    <p>
      {monthDay}
      {' '}
      {monthDayName}
    </p>
    <p>
      <img
        src={imgLink}
        width="100"
        height="100"
        alt="weather"
      />
    </p>
    <p>
      {temp.day}
    </p>
    <p>
      {temp.night}
    </p>
    <p>
      {weather}
    </p>
  </>
);
