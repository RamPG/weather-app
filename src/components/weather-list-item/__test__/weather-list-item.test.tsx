import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { WeatherListItem } from '../weather-list-item';
import { transformedDataDaily } from '../../../__mocks__/transformed-data';

describe('Weather list item', () => {
  afterEach(cleanup);
  it('Snapshot', () => {
    const { asFragment } = render(<WeatherListItem {...transformedDataDaily[0]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
