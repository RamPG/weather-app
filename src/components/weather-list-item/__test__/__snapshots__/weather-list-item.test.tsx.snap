// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather list item Snapshot 1`] = `
<DocumentFragment>
  <p>
    Friday
  </p>
  <p>
    14 August
  </p>
  <p>
    <img
      alt="weather"
      height="100"
      src="https://openweathermap.org/img/wn/04n@2x.png"
      width="100"
    />
  </p>
  <p>
    -249°C
  </p>
  <p>
    -259°C
  </p>
  <p>
    Clouds
  </p>
</DocumentFragment>
`;
